metadata_upload/metadata_upload_tutorial.Rmd

```diff
@@ -0,0 +1,623 @@
+
+In your home directory, create a `.pgpass` file and change the permissions to 600.
+
+```sh
+touch ~/.pgpass
+chmod 0600 ~/.pgpass
+```
+
+On a new line, enter the database connection parameters for the local instance of the BETY test database in the format:  
+
+`host:port:database:username:password`
+
+# Section 4: Set up for metadata upload
+
+When running this tutorial interactively:
+
+**You will need to update database connection parameters in this section on lines 117-120 to match those in your .pgpass file**
+
+**You will need to pass in the url to your google sheet in this section on line 138**
+
+## Load in packages
+
+```{r load-pack}
+# load in packages
+library(RPostgres)
+library(readxl)
+library(dplyr)
+library(googlesheets)
+```
+
+## Set options
+
+```{r set-option}
+options(scipen = 999)
+```
+
+## Connect to database; Provide connection parameters here (REQUIRED)
+
+_Please make sure the following database connection parameters (lines 117-120) match those that you have inputted in your .pgpass file. Make any needed changes._
+
+_The following connection parameters will be used when running this Rmd file from command line_
+
+```{r create-dbcon}
+# create database connection using parameters in .pgpass file
+dbcon <- dbConnect(RPostgres::Postgres(),
+                   dbname = 'bety',
+                   host = 'localhost',
+                   user = 'bety',
+                   port = 5432)
+```
+
+## Provide URL to google sheet here (REQUIRED)
+
+_this code chunk needs to be altered based on whether user is running tutorial interactively or Rmd file is run from command line_
+
+```{r user-url}
+# if user is running tutorial interactively :
+# leave lines 148 and 150 commented out
+# please provide URL to google sheet below on line 138
+# a sample URL has been provided for google sheet containing MAC season 8 metadata
+# please replace sample URL with your own
+url <- 'https://docs.google.com/spreadsheets/d/1c_5j7q3TO6gQ24KSopE-_LXA5HhfsUEqMupckGZAbo8/edit#gid=0' 
+########################################################
+# if Rmd file will be run from command line :
+# comment out line 138 above
+# uncomment lines 148 and 150
+# URL to google sheet must be passed in as command line arugment
+#args <- commandArgs(trailingOnly = TRUE)
+#url <- args[1]
+```
+
+Use URL provided by user to read directly from google sheets
+
+```{r read-gs}
+# get key from URL
+key <- extract_key_from_url(url)
+# register sheet 
+gs_obj <- key %>% gs_key(lookup = FALSE)
+## users sheet
+users <- gs_obj %>% gs_read(ws = 'users')
+## experiments sheet
+experiments <- gs_obj %>% gs_read(ws = 'experiments')
+## sites sheet
+sites <- gs_obj %>% gs_read(ws = 'sites')
+## treatments sheet
+treatments <- gs_obj %>% gs_read(ws = 'treatments')
+## cultivars sheet
+cultivars <- gs_obj %>% gs_read(ws = 'cultivars')
+## citations sheet
+citations <- gs_obj %>% gs_read(ws = 'citations')
+```
+
+## Reformat of data / Field additions
+
```

Comment:
we should expect that these are provided correctly in the spreadsheet, correct? Maybe explain that in some cases they may need to reformat data, but ideally they won't have to...